client/app/visualizations/counter/utils.js

```diff
@@ -0,0 +1,141 @@
+import { isNumber, isFinite, toString } from 'lodash';
+import numeral from 'numeral';
+
+// TODO: allow user to specify number format string instead of delimiters only
+// It will allow to remove this function (move all that weird formatting logic to a migration
+// that will set number format for all existing counter visualization)
```

Comment:
Can't we do this already without a migration by just translating the different settings into a format string for existing visualizations and use a string for new ones?

Reply:
It's the next thing I'd like to do. But I think it's better to do it as a separate feature (as implementation of getredash/redash#2558), but for a proper implementation in a single pass we need to do something with getredash/redash#3551 - it will allow to get rid of "Formatting String Prefix/Suffix" settings as well.

Reply:
:+1: